pyqir/pyqir/_native.pyi

```diff
@@ -21,8 +21,48 @@ class Attribute:
     """An attribute."""
 
     @property
-    def value(self) -> str:
-        """The value of the attribute as a string."""
+    def string_value(self) -> Optional[str]:
+        """The value of this attribute as a string, or `None` if this is not a string attribute."""
+        ...
+
+class AttributeDict:
+    """A group of attributes for a specific part of a function."""
+
+    def __contains__(self, item: str) -> bool:
+        """
+        Tests if an attribute is a member of the group.
+
+        :param item: The attribute kind.
+        :returns: True if the group has an attribute with the given kind.
+        """
+        ...
+    def __getitem__(self, key: str) -> Attribute:
```

Comment:
Do we need to document that this throws a key error if the key isn't found?

Reply:
No exceptions are documented currently.